pkg/watcher/metricsprovider.go

```diff
@@ -40,6 +41,7 @@ func init() {
 	}
 	EnvMetricProviderOpts.Address, ok = os.LookupEnv(MetricsProviderAddressKey)
 	EnvMetricProviderOpts.AuthToken, ok = os.LookupEnv(MetricsProviderTokenKey)
+	_, EnvMetricProviderOpts.InsecureSkipVerify = os.LookupEnv(InsecureSkipVerify)
```

Comment:
This will cause `INSECURE_SKIP_VERIFY: false` to be taken as "true". Remember we are checking the presence of `key:value` pair, not just key in the environment.